### PR TITLE
Vnet

### PR DESCRIPTION
**Pull request by larribas**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,4 +25,5 @@ _testmain.go
 
 tmp/
 .env
-test.go
\ No newline at end of file
+test.go
+*.swp
```
Comment (larribas) [inline .gitignore:29]: I'd recommend setting this in your global gitignore. Then it's on for all projects.

https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

File: opennebula/resource_template.go [hunk 2]
```diff
@@ -203,7 +203,7 @@ func resourceTemplateUpdate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	if d.HasChange("permissions") {
-		resp, err := changePermissions(intId(d.Id()), permission(d.Get("permissions").(string)), client)
+		resp, err := changePermissions(intId(d.Id()), permission(d.Get("permissions").(string)), client, "one.template.chmod")
```
Comment (larribas) [inline opennebula/resource_template.go:206]: nice

File: opennebula/resource_vnet.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+}
+
+type UserVnet struct {
+	Name                    string         `xml:"NAME"`
+	Id                      int            `xml:"ID"`
+	Uid                     int            `xml:"UID"`
+	Gid                     int            `xml:"GID"`
+	Uname                   string         `xml:"UNAME"`
+	Gname                   string         `xml:"GNAME"`
+	Permissions							*Permissions   `xml:"PERMISSIONS"`
```
Comment (larribas) [inline opennebula/resource_vnet.go]: I'll iterate on this. Just in general, in Go there is the `go fmt` command that would reformat your file to the Go standard

**Comment by tnosaj** [comment]
todo: write tests, think about multiple address_ranges per vnet, maybe make the reservations a bit more flexible

File: opennebula/resource_vnet.go [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+	d.Set("name", vn.Name)
+	d.Set("uid", vn.Uid)
+	d.Set("gid", vn.Gid)
+	d.Set("uname", vn.Uname)
+	d.Set("gname", vn.Gname)
+	d.Set("bridge", vn.Bridge)
+	//d.Set("ar", vn.Ar)
+	//d.Set("ip_start", vn.Ip_start)
+	//d.Set("ip_size", vn.Ip_size)
+	//d.Set("reservation-size", vn.Reservation_size)
```
Comment (larribas) [inline opennebula/resource_vnet.go]: should we remove the comments?

**Review by larribas — COMMENTED** [review]
Looks good!

Not much to comment on. I will see if I can find a way to avoid so much duplication in the future

**Pull request merged**